hardware_integration/src/can_hardware_interface.cpp

```diff
@@ -372,77 +345,73 @@ void CANHardwareInterface::can_thread_function()
 
 	while (threadsStarted)
 	{
-		std::unique_lock<std::mutex> lMutex(threadMutex);
-		threadConditionVariable.wait(lMutex);
-		CanHardware *pCANHardware;
+		std::unique_lock<std::mutex> threadLock(threadMutex);
+		threadConditionVariable.wait(threadLock);
 
 		if (threadsStarted)
 		{
 			for (std::uint32_t i = 0; i < hardwareChannels.size(); i++)
 			{
-				pCANHardware = hardwareChannels[i];
-
-				pCANHardware->receivedMessagesMutex.lock();
-				bool processNextMessage = (!pCANHardware->receivedMessages.empty());
-				pCANHardware->receivedMessagesMutex.unlock();
+				hardwareChannels[i]->receivedMessagesMutex.lock();
+				bool processNextMessage = (!hardwareChannels[i]->receivedMessages.empty());
+				hardwareChannels[i]->receivedMessagesMutex.unlock();
 
 				while (processNextMessage)
 				{
 					isobus::HardwareInterfaceCANFrame tempCanFrame;
 
-					pCANHardware->receivedMessagesMutex.lock();
-					tempCanFrame = pCANHardware->receivedMessages.front();
-					pCANHardware->receivedMessages.pop_front();
-					processNextMessage = (!pCANHardware->receivedMessages.empty());
-					pCANHardware->receivedMessagesMutex.unlock();
+					hardwareChannels[i]->receivedMessagesMutex.lock();
+					tempCanFrame = hardwareChannels[i]->receivedMessages.front();
+					hardwareChannels[i]->receivedMessages.pop_front();
+					processNextMessage = (!hardwareChannels[i]->receivedMessages.empty());
+					hardwareChannels[i]->receivedMessagesMutex.unlock();
 
-					rxCallbackMutex.lock();
+					rxCallbacksMutex.lock();
 					for (std::uint32_t j = 0; j < rxCallbacks.size(); j++)
 					{
 						if (nullptr != rxCallbacks[j].callback)
 						{
 							rxCallbacks[j].callback(tempCanFrame, rxCallbacks[j].parent);
 						}
 					}
-					rxCallbackMutex.unlock();
+					rxCallbacksMutex.unlock();
 				}
 			}
 
-			if (get_clear_can_lib_needs_update())
+			if (canLibNeedsUpdate)
```

Comment:
Where does this get reset to false?

Reply:
Oh yea good point, it didn't reset... Should be fixed now, I added `canLibNeedsUpdate = false;` directly underneath.